File: tutorials/tour-of-restate-typescript/src/part1/ticket_service.ts
/*
 * Copyright (c) 2024 - Restate Software, Inc., Restate GmbH
 *
 * This file is part of the Restate examples,
 * which is released under the MIT license.
 *
 * You can find a copy of the license in the file LICENSE
 * in the root directory of this repository or package or at
 * https://github.com/restatedev/examples/
 */

import * as restate from "@restatedev/restate-sdk";
// <start_import_sleep>
import { setTimeout } from "timers/promises";
// <end_import_sleep>

enum TicketStatus {
  Available,
  Reserved,
  Sold,
}

export const ticketDbRouter = restate.keyedRouter({
  // <start_reserve>
  async reserve(ctx: restate.KeyedContext){
    //bad-code-start
    await setTimeout(35000);
    //bad-code-end
    return true;
  },
  // <end_reserve>

  // <start_unreserve>
  async unreserve(ctx: restate.KeyedContext){
    return true;
  },
  // <end_unreserve>

  // <start_mark_as_sold>
  async markAsSold(ctx: restate.KeyedContext){
    return true;
  },
  // <end_mark_as_sold>
});

export const ticketServiceApi: restate.ServiceApi<typeof ticketDbRouter> = {
  path: "TicketService",
};

File: tutorials/tour-of-restate-java/src/main/java/dev/restate/tour/app/TicketService.java
/*
 * Copyright (c) 2024 - Restate Software, Inc., Restate GmbH
 *
 * This file is part of the Tour of Restate Java,
 * which is released under the MIT license.
 *
 * You can find a copy of the license in the file LICENSE
 * in the root directory of this repository or package or at
 * https://github.com/restatedev/examples/
 */

package dev.restate.tour.app;

import com.google.protobuf.BoolValue;
import dev.restate.sdk.ObjectContext;
import dev.restate.sdk.common.TerminalException;
import dev.restate.tour.generated.TicketServiceRestate;
import dev.restate.tour.generated.Tour.Ticket;

public class TicketService extends TicketServiceRestate.TicketServiceRestateImplBase {
    @Override
    public BoolValue reserve(ObjectContext ctx, Ticket request) throws TerminalException {
        return BoolValue.of(true);
    }

    @Override
    public void unreserve(ObjectContext ctx, Ticket request) throws TerminalException {
    }

    @Override
    public void markAsSold(ObjectContext ctx, Ticket request) throws TerminalException {
    }
}
